### PR TITLE
feat: modularity documentation

### PR DESCRIPTION
**Pull request by radoslaw-orlowski**
#### 📲 What

Changes in the documentation cased by introducing Java modularity.

#### ✅ Acceptance criteria Checklist

- [x] Code peer reviewed?
- [x] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: docs/workloads/azure/backend/java/scaffolding_java.md [hunk 1]
```diff
@@ -22,30 +22,43 @@ import ResponsivePlayer from "../../../../../src/pages/Components/ResponsivePlay
...
+by cloning one of the workload repositories and then installing the project as a Maven archetype locally; from there, we can create a new project by using
 that archetype as a template. To do so we do:
 
-1. Clone any of the STACKS Java project (simple-api, stacks-java-cqrs, stacks-java-cqrs-events)
+1. Clone any of the Stacks Java Workloads (`simple-api`, `stacks-java-cqrs` or `stacks-java-cqrs-events`)
 2. Navigate to the `<directory you cloned the project into>/java` in the terminal
 3. Then issue the following Maven commands, using the included wrapper:
-   1. ``./mvnw archetype:create-from-project`` - To create the archetype
-   2. `` cd target/generated-sources/archetype`` - Navigate to the folder it was crated in
+   1. ``./mvnw archetype:create-from-project -DpropertyFile=archetype.properties`` - To create the archetype
```
Comment (radoslaw-orlowski) [inline docs/workloads/azure/backend/java/scaffolding_java.md:36]: @guillermogutierrez / @johnsall : is this still required?

File: versions.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- then reference like: <tag>{siteVariables.variableName}</tag>
+ */
+export const siteVariables = {
+    javaModuleParentVersion: 'parentversion',
+    javaCommonsVersion: 'commonsversion',
+    javaApiVersion: 'apiversion',
+    javaMessagingVersion: 'messagingversion',
+    javaCqrsVersion: 'cqrsversion',
+    javaAzureCosmosVersion: 'cosmosversion',
+    javaAzureServicebusVersion: 'servicebusversion'
```
Comment (radoslaw-orlowski) [inline versions.js:15]: @johnsall / @guillermogutierrez : do you mind providing the current list of the used modules? These are injected into the individual modules' `pom.xml` `dependency` version sections.

Comment (radoslaw-orlowski) [reply versions.js]: edit: currently all are set to `1.0.0`.

File: docs/workloads/azure/backend/java/scaffolding_java.md (hunk 1 above)
Comment (guillermogutierrez) [reply docs/workloads/azure/backend/java/scaffolding_java.md:36]: Yes it is, the properties file includes file extensions and exclusions

**Review by guillermogutierrez — APPROVED** [review]

**Pull request merged**